packages/strapi-utils/lib/models.js

```diff
@@ -418,96 +436,262 @@ module.exports = {
+      );
     }
 
     // Remove the source params (that can be sent from the ctm plugin) since it is not a filter
     if (params.source) {
       delete params.source;
     }
 
-    const model = entity.toLowerCase();
+    const modelName = entity.toLowerCase();
+    const models = this.getStrapiModels();
+    const model = models[modelName];
 
-    const models = _.assign(_.clone(strapi.models), Object.keys(strapi.plugins).reduce((acc, current) => {
-      _.assign(acc, _.get(strapi.plugins[current], ['models'], {}));
-      return acc;
-    }, {}));
+    if (!model) {
+      throw new Error(`The model ${modelName} can't be found.`);
+    }
 
-    if (!models.hasOwnProperty(model)) {
-      return this.log.error(`The model ${model} can't be found.`);
+    if (!model.orm) {
+      throw new Error(
+        `Impossible to determine the ORM used for the model ${modelName}.`
+      );
     }
 
-    const client = models[model].client;
-    const connector = models[model].orm;
+    const hook = strapi.hook[model.orm];
+    const convertor = hook.load().getQueryParams;
+    const postProcessValue = hook.load().postProcessValue || _.identity;
 
-    if (!connector) {
-      throw new Error(`Impossible to determine the ORM used for the model ${model}.`);
-    }
+    return {
+      models,
+      model,
+      hook,
+      convertor,
+      postProcessValue,
+    };
+  },
 
-    const convertor = strapi.hook[connector].load().getQueryParams;
-    const convertParams = {
-      where: {},
-      sort: '',
-      start: 0,
-      limit: 100
+  getStrapiModels: function() {
+    return {
+      ...strapi.models,
+      ...Object.keys(strapi.plugins).reduce(
+        (acc, pluginName) => ({
+          ...acc,
+          ..._.get(strapi.plugins[pluginName], 'models', {}),
+        }),
+        {}
+      ),
     };
+  },
 
-    _.forEach(params, (value, key)  => {
-      let result;
-      let formattedValue;
-      let modelAttributes = models[model]['attributes'];
-      let fieldType;
-      // Get the field type to later check if it's a string before number conversion
-      if (modelAttributes[key]) {
-        fieldType = modelAttributes[key]['type'];
-      } else {
-        // Remove the filter keyword at the end
-        let splitKey = key.split('_').slice(0,-1);
-        splitKey = splitKey.join('_');
-
-        if (modelAttributes[splitKey]) {
-          fieldType = modelAttributes[splitKey]['type'];
+  splitPrimitiveAndRelationValues: function(_query) {
+    const result = _.reduce(
+      _query,
+      (acc, value, key) => {
+        if (_.startsWith(key, '_')) {
+          acc[key] = value;
+        } else if (!_.includes(key, '.')) {
+          acc.where[key] = value;
+        } else {
+          _.set(acc.relations, this.injectRelationInKey(key), value);
         }
+        return acc;
+      },
+      {
+        where: {},
+        relations: {},
```

Comment:
I probably need to change this to something more generic which will not conflict with someone's field name, I was thinking of something like `strapi__relations`